Vagrantfile

```diff
@@ -21,13 +90,34 @@ Vagrant.configure(2) do |config|
   vm_memory = ENV.fetch('VM_MEMORY', 10 * 1024).to_i
   vm_cpus = ENV.fetch('VM_CPUS', 4).to_i
 
+  vb_net_config = {}
+  if bridged_net?
+    vb_net_config[:use_dhcp_assigned_default_route] = true
+  else
+    # Use dhcp if VAGRANT_DHCP is set. This only applies to NAT networking, as
+    # bridged networking uses type: bridged (even though the virtual interface still
+    # gets its IP from dhcp. If not using dhcp, the VM will use the 192.168.77.77 IP
+    if ENV.fetch("VAGRANT_DHCP", false)
+      vb_net_config[:type] = "dhcp"
+    else
+      vb_net_config[:ip] = "192.168.77.77"
+    end
+  end
+  # Create a clone of this, otherwise it gets mutated in both providers' sections
+  libvirt_net_config = vb_net_config.clone
```

Comment:
I _think_ a helper function to construct the base net config would be slightly better (because it never pretends to be the vbox net config). But not by much.